app/javascript/article-form/components/EditorBody.jsx

```diff
@@ -74,19 +80,20 @@ export const EditorBody = ({
       className="crayons-article-form__body drop-area text-padding"
     >
       <Toolbar version={version} />
-
-      <Textarea
-        className="crayons-textfield crayons-textfield--ghost crayons-article-form__body__field ff-monospace fs-l"
-        id="article_body_markdown"
-        aria-label="Post Content"
-        placeholder="Write your post content here..."
-        value={defaultValue}
-        onInput={onChange}
-        onFocus={(_event) => {
-          switchHelpContext(_event);
-        }}
-        name="body_markdown"
+      <MentionAutocompleteTextArea
         ref={textAreaRef}
+        fetchSuggestions={(username) => fetchSearch('usernames', { username })}
+        inputProps={{
```

Comment:
Refactor (blocking): Instead of `inputProps`, add the props defined in `inputProps` as top level props to `<MentionAutocompleteTextArea />`. We can still spread these props like you currently do, e.g.

**Not the `...inputProps` here in the forward red as well as no default value for `inputProps`. If you spread undefined, it returns an empty object.**

![image](https://user-images.githubusercontent.com/833231/112331817-a4524880-8c8f-11eb-83da-b9dda5bfb5b4.png)

```jsx
export const MentionAutocompleteTextArea = forwardRef(
  ({ replaceElement, fetchSuggestions, ...inputProps }

...

return (
      <Fragment>
        <div aria-live="polite" class="screen-reader-only">
          {ariaHelperText}
        </div>
        <Combobox
          id="combobox-container"
          onSelect={handleSelect}
          className="crayons-autocomplete"
        >
          <ComboboxInput
            {...inputProps}
            ref={mergeRefs([inputRef, forwardedRef])}
            value={textContent}
            data-mention-autocomplete-active="true"
            as="textarea"
            autocomplete={false}
            onChange={(e) => {
              inputProps.onChange?.(e);
              handleTextInputChange(e);
            }}
          />
          ...
```

Reply:
Refactor (blocking): One other thing to note is that `{...inputProps}` will spread all those props including the potential `inputProps.onChange`. It won't affect anything as the last `onChange` wins, but it should be filtered out.

```jsx
const { onChange, ...comboBoxProps } = inputProps;


return (
      <Fragment>
        <div aria-live="polite" class="screen-reader-only">
          {ariaHelperText}
        </div>
        <Combobox
          id="combobox-container"
          onSelect={handleSelect}
          className="crayons-autocomplete"
        >
          <ComboboxInput
            {...comboBoxProps}
            ref={mergeRefs([inputRef, forwardedRef])}
            value={textContent}
            data-mention-autocomplete-active="true"
            as="textarea"
            autocomplete={false}
            onChange={(e) => {
              onChange?.(e);
              handleTextInputChange(e);
            }}
          />
          ...
```